src/coreclr/jit/emitxarch.cpp

```diff
@@ -609,6 +609,54 @@ bool emitter::AreUpper32BitsZero(regNumber reg)
     return false;
 }
 
+//------------------------------------------------------------------------
+// AreUpper32BitsFFFF: check if some previously emitted
+//     instruction set the upper 32 bits of reg to 0xFFFF.
+//
+// Arguments:
+//    reg - register of interest
+//
+// Return Value:
+//    true if previous instruction upper 32 bits are 0xFFFF.
+//    false if it did not, or if we can't safely determine.
+bool emitter::AreUpper32BitsFFFF(regNumber reg)
+{
+    // Only allow GPRs.
+    // If not a valid register, then return false.
+    if (!genIsValidIntReg(reg))
```

Comment:
Should this be an assert instead? I'd think we wouldn't want to be calling this for simd registers in the first place.

Reply:
We already have this pattern for `AreUpper32BitsZero`.